src/pkg/term/colorizer.go

```diff
@@ -58,84 +131,205 @@ func output(w *termenv.Output, c Color, msg string) (int, error) {
 
 func outputf(w *termenv.Output, c Color, format string, v ...any) (int, error) {
 	line := fmt.Sprintf(format, v...)
-	if len(line) == 0 || (line[len(line)-1] != '\n' && line[len(line)-1] != '\r') {
-		line += "\n" // add newline, like log.Printf
-	}
+	line = ensureNewline(line)
 	return output(w, c, line)
 }
 
-func Fprint(w *termenv.Output, c Color, v ...any) (int, error) {
-	return output(w, c, fmt.Sprint(v...))
+func ensureNewline(s string) string {
+	if len(s) == 0 || (s[len(s)-1] != '\n' && s[len(s)-1] != '\r') {
+		return s + "\n"
+	}
+	return s
 }
 
-func Fprintln(w *termenv.Output, c Color, v ...any) (int, error) {
-	return output(w, c, fmt.Sprintln(v...))
+func (t *Term) Printc(c Color, v ...any) (int, error) {
+	return output(t.stdout, c, fmt.Sprint(v...))
 }
 
-func Fprintf(w *termenv.Output, c Color, format string, v ...any) (int, error) {
-	return output(w, c, fmt.Sprintf(format, v...))
+func (t *Term) Printlnc(c Color, v ...any) (int, error) {
+	return output(t.stdout, c, ensureNewline(fmt.Sprintln(v...)))
 }
 
-func Print(c Color, v ...any) (int, error) {
-	return Fprint(Stdout, c, v...)
+func (t *Term) Printfc(c Color, format string, v ...any) (int, error) {
+	return outputf(t.stdout, c, format, v...)
 }
 
-func Println(c Color, v ...any) (int, error) {
-	return Fprintln(Stdout, c, v...)
+func (t *Term) Print(v ...any) (int, error) {
+	return fmt.Fprint(t.stdout, v...)
 }
 
-func Printf(c Color, format string, v ...any) (int, error) {
-	return Fprintf(Stdout, c, format, v...)
+func (t *Term) Println(v ...any) (int, error) {
+	return fmt.Fprint(t.stdout, ensureNewline(fmt.Sprintln(v...)))
 }
 
-func Debug(v ...any) (int, error) {
-	if !DoDebug {
+func (t *Term) Printf(format string, v ...any) (int, error) {
+	return fmt.Fprintf(t.stdout, format, ensureNewline(fmt.Sprint(v...)))
+}
+
+func (t *Term) Debug(v ...any) (int, error) {
+	if !t.debug {
 		return 0, nil
 	}
-	return Fprintln(Stderr, DebugColor, v...)
+	return t.Printlnc(DebugColor, v...)
```

Comment:
We've been starting all debug text with a " - ". It may make sense to just insert it here rather than depend on the caller adding it.

Reply:
Good point, added the prefix check and does a automatic prefix if it is not already there, with test cases.